### PR TITLE
[#5983] Fix unit tests in nacos-config

### LINKED ISSUE #5983: Fix unit tests in nacos-config
parent ISSUE: #5976

### PR DESCRIPTION
**Pull request by realJackSun**
fix #5983 

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/test/java/com/alibaba/nacos/config/server/controller/HealthControllerUnitTest.java [hunk 1]
```diff
@@ -47,15 +57,31 @@
...
     @Before
-    public void setUp() throws Exception {
+    public void setUp() {
+        EnvUtil.setEnvironment(new StandardEnvironment());
+        Map<String, Object> infos = new HashMap<>();
+        infos.put("addressServerHealth", true);
+        when(memberLookup.info()).thenReturn(infos);
+        when(memberManager.getLookup()).thenReturn(memberLookup);
+        when(servletContext.getContextPath()).thenReturn("/nacos");
+        ReflectionTestUtils.setField(healthController, "memberManager", memberManager);
```
Comment (realJackSun) [inline config/src/test/java/com/alibaba/nacos/config/server/controller/HealthControllerUnitTest.java:77]: healthController中的memberManager没有被注入，所以这里mock一个注入。

File: config/src/test/java/com/alibaba/nacos/config/server/service/ConfigChangePublisherTest.java [hunk 2]
```diff
@@ -84,6 +84,9 @@ public void onEvent(Event event) {
         Thread.sleep(2000);
         Assert.assertNotNull(reference.get());
         reference.set(null);
+    
+        EnvUtil.setIsStandalone(true);
+        PropertyUtil.setEmbeddedStorage(true);
```
Comment (realJackSun) [inline config/src/test/java/com/alibaba/nacos/config/server/service/ConfigChangePublisherTest.java:89]: 这个修改的动机是，这个单测会通过EnvUtil.setIsStandalone(false);破坏原来的EnvUtil的字段值。
这里两行代码，将EnvUtil重置为原先的值。

File: config/src/test/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#
```
Comment (realJackSun) [inline config/src/test/resources/application.properties:1]: 加上这个配置文件的意义在于，没有这个配置文件，DumpServiceTest的前置@SpringBootTest就启动不了。

File: config/src/test/java/com/alibaba/nacos/config/server/service/ConfigChangePublisherTest.java (hunk 2 above)
Comment (KomachiSion) [reply config/src/test/java/com/alibaba/nacos/config/server/service/ConfigChangePublisherTest.java:89]: 在Before方法和After方法中去还原。否则异常情况下仍然会导致不会重制。

File: config/src/test/java/com/alibaba/nacos/config/server/service/dump/DumpServiceTest.java [hunk 4]
```diff
@@ -19,12 +19,14 @@
...
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 import org.springframework.test.context.web.WebAppConfiguration;
 
 @RunWith(SpringJUnit4ClassRunner.class)
 @SpringBootTest
+@SpringBootApplication(scanBasePackages = "com.alibaba.nacos")
```
[the remark this answers is not in the document]
Comment (KomachiSion) [reply config/src/test/java/com/alibaba/nacos/config/server/service/dump/DumpServiceTest.java:29]: 无法mock？

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: auth/src/main/java/com/alibaba/nacos/auth/common/AuthConfigs.java [hunk 5]
```diff
@@ -34,6 +35,7 @@
  * @since 1.2.0
  */
```
Comment (realJackSun) [inline auth/src/main/java/com/alibaba/nacos/auth/common/AuthConfigs.java:36]: 这个不需要，可以研究一下

**Comment by KomachiSion** [comment]
Please pull newest codes, which fix ci problem.

**Review by KomachiSion — APPROVED** [review]

**Review by KomachiSion — APPROVED** [review]

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**